miden-tx/src/testing/tx_context.rs

```diff
@@ -211,33 +216,369 @@ impl TransactionContextBuilder {
         self
     }
 
-    /// Populates input and expected notes.
-    pub fn with_mock_notes(self, asset_preservation: AssetPreservationStatus) -> Self {
-        let (mut input_notes, output_notes) = mock_notes(&self.assembler);
-
-        let consumed_note_5 = input_notes.pop().unwrap();
-        let consumed_note_4 = input_notes.pop().unwrap();
-        let consumed_note_3 = input_notes.pop().unwrap();
-        let consumed_note_2 = input_notes.pop().unwrap();
-        let consumed_note_1 = input_notes.pop().unwrap();
-
-        let notes = match asset_preservation {
-            AssetPreservationStatus::TooFewInput => vec![consumed_note_1],
-            AssetPreservationStatus::Preserved => {
-                vec![consumed_note_1, consumed_note_2]
-            },
-            AssetPreservationStatus::PreservedWithAccountVaultDelta => {
-                vec![consumed_note_1, consumed_note_2, consumed_note_5]
-            },
-            AssetPreservationStatus::TooManyFungibleInput => {
-                vec![consumed_note_1, consumed_note_2, consumed_note_3]
-            },
-            AssetPreservationStatus::TooManyNonFungibleInput => {
-                vec![consumed_note_1, consumed_note_2, consumed_note_4]
-            },
-        };
-
-        self.input_notes(notes).expected_notes(output_notes)
+    /// Creates a new output [Note] for the transaction corresponding to this context.
+    fn add_output_note(
+        &mut self,
+        inputs: impl IntoIterator<Item = Felt>,
+        assets: impl IntoIterator<Item = Asset>,
+    ) -> Note {
+        let note = NoteBuilder::new(self.account.id(), &mut self.rng)
+            .note_inputs(inputs)
+            .expect("The inputs should be valid")
+            .add_assets(assets)
+            .build(&self.assembler)
+            .expect("The note details should be valid");
+
+        self.expected_output_notes.push(note.clone());
+        note
+    }
+
+    fn input_note_simple(
+        &mut self,
+        sender: AccountId,
+        assets: impl IntoIterator<Item = Asset>,
+        inputs: impl IntoIterator<Item = Felt>,
+    ) -> Note {
+        NoteBuilder::new(sender, ChaCha20Rng::from_seed(self.rng.gen()))
+            .note_inputs(inputs)
+            .unwrap()
+            .add_assets(assets)
+            .build(&self.assembler)
+            .unwrap()
+    }
```

Comment:
NOTE: I'm not sure any of these notes are valid. The default code of a note is empty, literally `begin end`, meaning the `assets` are not added to the account running the tx kernel nor added to a new output note. These should fail on the epilogue check, when the asset vault from the inputs+account is compared against the outputs+account. I believe tests are not failing because we are not testing the `main.masm`.

With that said, I didn't dig through further. I would need to read all the tests, and figure out how they are using the notes, to know what the fix is (that is, if my guess above is even correct). Figuring out what these functions are actually used for needs more work.